### ISSUE TITLE
GradientBoostingClassifier converter predicted probabilities don't match with Scikit predicitons. 

### ISSUE THREAD

**Issue by prabhat00155** [onnx/sklearn-onnx#184]
data = load_iris()
X = data.data
y = data.target

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.5, random_state=42)
model = GradientBoostingClassifier(loss='deviance', learning_rate=0.21, n_estimators=10, subsample=1.0,
                                   criterion='friedman_mse', min_samples_split=2, min_samples_leaf=1,
                                   min_weight_fraction_leaf=0.0, max_depth=3, min_impurity_decrease=0.0,
                                   min_impurity_split=None, init=None, random_state=None, max_features=None,
                                   verbose=0, max_leaf_nodes=None, warm_start=False, presort='auto',
                                   validation_fraction=0.1, n_iter_no_change=None, tol=0.0001).fit(X_train, y_train)

onnx_model = convert_sklearn(model, 'lr', [('input', FloatTensorType(X_test.shape))])
save_model(onnx_model, 'lr1.onnx')
sess = InferenceSession('lr1.onnx')
res = sess.run(None, input_feed={'input': X_test.astype(np.float32)})

r1 = np.mean(np.isclose(model.predict_proba(X_test)[:10], list(map(lambda x:
                                                         list(map(lambda y: x[y], x)), res[1][:10])), atol=1e-4))
r2 = np.mean(res[0] == model.predict(X_test))
print(f'{r1} {r2}')

Result: 0.03333333333333333 1.0
